libs/ml/include/ml/optimisation/optimiser.hpp

```diff
@@ -157,7 +186,7 @@ typename T::Type Optimiser<T, C>::Run(std::vector<ArrayType> const &data, ArrayT
   UpdateLearningRate();
   epoch_++;
 
-  return loss_sum / static_cast<DataType>(step);
+  return loss_sum;
```

Comment:
removed loss normalisation? is this by mistake?

Reply:
or are the loss functions now responsible for the normalisation?

Reply:
Because you call it on a batch, Criterion returns average of batch now. It will be rewritten anyway In next PR, where criterions are ops and will return Tensor of errors instead.